official/vision/beta/ops/augment.py

```diff
@@ -1267,3 +1306,240 @@ def distort(self, image: tf.Tensor) -> tf.Tensor:
+        rectangle. Defaults to 0.3.
+      max_aspect ([type], optional): Maximum aspect rate of the random
+        erasing rectangle. Defaults to None.
+      min_count (int, optional): Minimum number of erased
+        rectangles. Defaults to 1.
+      max_count (int, optional):  Maximum number of erased
+        rectangles. Defaults to 1.
+      trials (int, optional): Maximum number of trials to randomly sample a
+        rectangle that fulfills constraint. Defaults to 10.
+    """
+    self._probability = probability
+    self._min_area = float(min_area)
+    self._max_area = float(max_area)
+    self._min_log_aspect = math.log(min_aspect)
+    self._max_log_aspect = math.log(max_aspect or 1 / min_aspect)
+    self._min_count = min_count
+    self._max_count = max_count
+    self._trials = trials
+
+  def distort(self, image: tf.Tensor) -> tf.Tensor:
+    """Applies RandomErasing to single `image`.
+
+    Args:
+      image (tf.Tensor): Of shape [height, width, 3] representing an image.
+
+    Returns:
+      tf.Tensor: The augmented version of `image`.
+    """
+    uniform_random = tf.random.uniform(shape=[], minval=0., maxval=1.0)
+    mirror_cond = tf.less(uniform_random, .5)
+    tf.cond(mirror_cond, self._erase, lambda: image)
+    return image
+
+  @tf.function
+  def _erase(self, image: tf.Tensor) -> tf.Tensor:
+    count = self._min_count if self._min_count == self._max_count else \
+        tf.random.uniform(shape=[], minval=int(self._min_count),
+                          maxval=int(self._max_count - self._min_count + 1),
+                          dtype=tf.int32)
+
+    image_height = tf.shape(image)[0]
+    image_width = tf.shape(image)[1]
+    area = tf.cast(image_width * image_height, tf.float32)
+
+    for _ in range(count):
+      for _ in range(self._trials):
+        erase_area = tf.random.uniform(shape=[],
+                                       minval=area * self._min_area,
+                                       maxval=area * self._max_area)
+        aspect_ratio = tf.math.exp(tf.random.uniform(
+            shape=[], minval=self._min_log_aspect,
+            maxval=self._max_log_aspect))
+
+        half_height = tf.cast(tf.math.round(tf.math.sqrt(
+            erase_area * aspect_ratio) / 2), dtype=tf.int32)
+        half_width = tf.cast(tf.math.round(tf.math.sqrt(
+            erase_area / aspect_ratio) / 2), dtype=tf.int32)
+
+        if 2 * half_height < image_height and 2 * half_width < image_width:
+          center_height = tf.random.uniform(
+              shape=[], minval=0, maxval=int(image_height - 2 * half_height),
+              dtype=tf.int32)
+          center_width = tf.random.uniform(
+              shape=[], minval=0, maxval=int(image_width - 2 * half_width),
+              dtype=tf.int32)
+
+          image = _fill_rectangle(image, center_width, center_height,
+                                  half_width, half_height, replace=None)
+
+          break
+
+    return image
+
+
+class MixupAndCutmix:
```

Comment:
Can you separate them to two classes?

Reply:
The only difficulty in separating Cutmix and Mixup is located in [`distort`](https://github.com/tensorflow/models/pull/10187/files#diff-43bd3b3bd437704043e9d5bfa065c61b03e1488f5e9366e8820fe7e8b6b23caeR1473). Here, I randomly choose either Mixup or Cutmix. 

Hence, if I separate them into two classes I would need to either (1) move this "choose on of" feature to [the `image_classification` task](https://github.com/tensorflow/models/pull/10187/files#diff-5abe8251e133688b5f899bf7b14f718f98bb82fc1bc905ce28b51e393d3573b0R112) or (2) to add a "wrapper" here in the `augment.py`. What do you think is the best solution?

Reply:
Let's keep it as it is in the first version.